debezium-core/src/main/java/io/debezium/function/BufferedBlockingConsumer.java

```diff
@@ -42,27 +43,31 @@ public default void flush() throws InterruptedException {
      * and buffer the latest.
      * <p>
      * The resulting consumer is not threadsafe.
-     * 
+     *
      * @param delegate the delegate to which values should be flushed; may not be null
      * @return the blocking consumer that buffers a single value at a time; never null
      */
     public static <T> BufferedBlockingConsumer<T> bufferLast(BlockingConsumer<T> delegate) {
         return new BufferedBlockingConsumer<T>() {
-            private T last;
+
+            private final AtomicReference<T> last = new AtomicReference<>();
 
             @Override
             public void accept(T t) throws InterruptedException {
-                if (last != null) delegate.accept(last);
-                last = t;
+                T previous = last.getAndSet(t);
```

Comment:
This code changes the order of execution in comparison to the previous code thus I opted for `synchronized`. I am not sure if the order matters or not

Reply:
Isn't it only changing the order of execution to be an atomic swap?

Reply:
> What is different about this approach is that it does not prevent the accept and flush from being called concurrently.

Right, it was my thinking that the calling code needs to make sure that it only calls `flush()` after the `accept()` calls.